src/SignalRServiceExtension/Config/ServiceManagerStore.cs

```diff
@@ -3,67 +3,59 @@
-            {
-                this.transportType = transport;
-            }
-            else
-            {
-                this.transportType = ServiceTransportType.Transient;
-                logger.LogWarning($"Unsupported service transport type: {serviceTransportTypeStr}. Use default {transportType} instead.");
-            }
-        }
-
-        public IServiceHubContextStore GetOrAddByConfigurationKey(string configurationKey)
-        {
-            string connectionString = configuration[configurationKey];
-            return GetOrAddByConnectionString(connectionString);
         }
 
-        public IServiceHubContextStore GetOrAddByConnectionString(string connectionString)
+        public IServiceHubContextStore GetOrAddByConnectionStringKey(string connectionStringKey)
         {
-            return store.GetOrAdd(connectionString, CreateHubContextStore);
-        }
-
-        // test only
-        public IServiceHubContextStore GetByConfigurationKey(string configurationKey)
-        {
-            string connectionString = configuration[configurationKey];
-            return store.ContainsKey(connectionString) ? store[connectionString] : null;
+            if (string.IsNullOrWhiteSpace(connectionStringKey))
+            {
+                throw new ArgumentException($"'{nameof(connectionStringKey)}' cannot be null or whitespace", nameof(connectionStringKey));
+            }
+            return store.GetOrAdd(connectionStringKey, CreateHubContextStore);
         }
 
-        private IServiceHubContextStore CreateHubContextStore(string connectionString)
+        //test only
+        public IServiceHubContextStore GetByConfigurationKey(string connectionStringKey)
         {
-            var serviceManager = CreateServiceManager(connectionString);
-            return new ServiceHubContextStore(serviceManager, loggerFactory);
+            return store.ContainsKey(connectionStringKey) ? store[connectionStringKey] : null;
         }
 
-        private IServiceManager CreateServiceManager(string connectionString)
+        private IServiceHubContextStore CreateHubContextStore(string connectionStringKey)
         {
-            return new ServiceManagerBuilder().WithOptions(o =>
-            {
-                o.ConnectionString = connectionString;
-                o.ServiceTransportType = transportType;
-            }).WithCallingAssembly().Build();
+            return new ServiceCollection().AddSignalRServiceManager()
+                .WithAssembly(Assembly.GetExecutingAssembly())
+                .SetupOptions<ServiceManagerOptions,OptionsSetup>(new OptionsSetup(configuration, loggerFactory, connectionStringKey))
+                .PostConfigure<ServiceManagerOptions>(o =>
+                {
+                    if (string.IsNullOrWhiteSpace(o.ConnectionString))
+                    {
+                        throw new InvalidOperationException(ErrorMessages.EmptyConnectionStringErrorMessageFormat);
```

Comment:
Add the info about attribute as original error message

Reply:
We don't know the attribute here. It can be any attribute.